### PR TITLE
zip .env files before encryption

### PR DESCRIPTION
**Pull request by sastels**
## What are you changing?
- Makefile

## Provide some background on the changes

Our production `.env` file is at the size limit of what may be encrypted using `aws kms encrypt`.

We could use secrets manager (see, for example https://jashby2.medium.com/injecting-secrets-from-aws-secrets-manager-into-kubernetes-9aad3c98e078#:~:text=Grab%20a%20list%20of%20secrets,cluster%20as%20native%20k8s%20secrets) though the process is a bit cumbersome. As a quick fix, this PR zips the file before encrypting, reducing the size by about 40%.

Note that I haven't included the new encrypted files `.env.zip.enc.aws` yet (nor deleted the `.env.enc.aws` files)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,48 +1,56 @@
...
 
-diff-staging: decrypt-staging
+diff-staging: decrypt-staging decrypt-previous-staging
 	@cd env/staging &&\
-	curl -L -o .previous.env.enc.aws https://github.com/cds-snc/notification-manifests/blob/main/env/staging/.env.enc.aws?raw=true > /dev/null 2>&1 &&\
-	aws kms decrypt --ciphertext-blob fileb://.previous.env.enc.aws --output text --query Plaintext --region ca-central-1 | base64 --decode > .previous.env &&\
 	diff .previous.env .env
 
-diff-production: decrypt-production
+diff-production: decrypt-production decrypt-previous-production
```
Comment (jimleroyer) [inline Makefile:41]: Good improvement. 👍

**Comment by jimleroyer** [comment]
Is this ready for review? (the PR is still in draft)

**Comment by sastels** [comment]
Didn't want to push the new encrypted zip files if someone saw a glaring error somewhere. Will do that now.

**Comment by sastels** [comment]
ok, right now I've got both old and new encrypted files, will merge this and we can see if it works, and then clean up in a subsequent PR.

**Comment by sastels** [comment]
I have some gc articles env vars to add, I'll see it it works using the new zipped process in a PR after this one.

**Review by jimleroyer — APPROVED** [review]

**Pull request merged**